common/windows/win32_window.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WIN32_WINDOW_H_
+#define WIN32_WINDOW_H_
+
+#include <Windows.h>
+#include <Windowsx.h>
+
+#include <functional>
+#include <memory>
+#include <string>
+
+// A class abstraction for a high DPI aware Win32 Window.  Intended to be
+// inherited from by classes that wish to specialize with custom
+// rendering and input handling
+class Win32Window {
+ public:
+  Win32Window();
+  ~Win32Window();
+
+  // Creates and shows a win32 window with |title| and position and size using
+  // |x|, |y|, |width| and |height|
+  bool CreateAndShow(const char *title, const unsigned int x,
+                     const unsigned int y, const unsigned int width,
+                     const unsigned int height);
```

Comment:
There is RECT which is top left right bottom.  A bit annoying as you can't express what you really want (width and height).  Can switch to that if you like but it will create even more boilerplate windowing code as we will need to reconstruct width.& height to correctly preserve DPI handling.